zegami_sdk/util.py

```diff
@@ -108,12 +107,42 @@ def _auth_post(self, url, body, return_response=False, **kwargs):
 
 def _auth_put(self, url, body, return_response=False, **kwargs):
     """Syncronous PUT request. Used as standard over async currently.
-
     If return_response == True, the response object is returned rather than
     its .json() output.
-
     Any additional kwargs are forwarded onto the requests.put().
     """
     r = requests.put(url, body, headers=self.headers, **kwargs)
     self._check_status(r, is_async_request=False)
     return r if return_response else r.json()
+
+
+def _create_singed_blob_storage(self, workspace_id):
+    """Create a signed blob storage.
+
+    Returns:
+        [str]: blob storage url
+        [str]: blob storage id
+    """
+    blob_url = f'{self.HOME}/{self.API_1}/project/{workspace_id}/signed_blob_url'
+    blob_id = str(uuid.uuid4())
+    response = self._auth_post(blob_url, body=None, json={"ids": [blob_id]}, return_response=True)
+    try:
+        data = response.json()
+        url = data[blob_id]
+        return url, blob_id
+    except Exception as ex:
+        print(ex)
+
+
+def _upload_to_signed_blob_storage(data, url, mime_type, headers=None, **kwargs):
+    """Upload data to an already create blob storage."""
+    if url.startswith("/"):
+        url = f'https://storage.googleapis.com{url}'
+    headers = {'Content-Type': mime_type}
+    if 'windows.net' in url:
+        headers['x-ms-blob-type'] = 'BlockBlob'
```

Comment:
worth a comment here explaining that azure blob storage requires this header and maybe a link to docs